hooks/tests/.claude/hooks/tests/test_speculation_detection.sh

```
@@ -0,0 +1,45 @@
+#\!/bin/bash
+# Red-Green Test for speculation detection hook
+
+HOOK_SCRIPT="../detect_speculation.sh"
+
+echo "=== RED-GREEN TEST: Speculation Detection ==="
+
+# TEST 1 (RED): Should FAIL - Contains speculation
+echo "Test 1 (RED): Testing speculation detection..."
+TEST_TEXT="I'll wait for the /commentr command to complete and provide its results before proceeding with any additional work."
+
+echo "$TEST_TEXT"  < /dev/null |  bash "$HOOK_SCRIPT"
```

Shellcheck (0.10.0)

[warning] Redirecting to 'echo', a command that doesn't read stdin. Did you want 'cat' instead?

(SC2217)